### PR TITLE
Adding in some missing documentation to the MRTK3 source.

### PR DESCRIPTION
**Pull request by AMollis**
## Overview
Adding in some missing documentation to the MRTK3 source.  

This also replaces some `ref` parameters with `in` parameters, in cases where it was obvious that only "const refs" were needed.  This also removed the `InterpolationType.Hermite` since it was never handled or used.

The following were documentated:
  - [x] `CameraSettings` class
  - [x] `LabelWidthAttribute` class
  - [x] `MathExtensions.PowerOfTwoGreate=rThanOrEqualTo`
  - [x] `MathExtensions.CubicToLinearIndex`
  - [x] `MathExtensions.LinearToCubicIndex`
  - [x] `MathExtensions.ClampComponentWise`
  - [x] `RayStep` struct and some functions
  - [x] `StatefulInteractable` class
  - [x] `SystemInterfaceType` constructors
  - [x] `SystemType` public functions
  - [x] `TimedFlag` class definition
  - [x] `UXBindingProfileTemplate` properties
  - [x] `VectorExtensions` some functions
  - [x] `TextToSpeechSubsystem.Provider` class definition
  - [x] `BezierInertia` class definition, serialized field tooltips
  - [x] `FuzzyGazeInteractor.GazeRaycastHitResult` struct definition and public members
  - [x] `HandJointInteractor` public method
  - [x] `HandshapeTypes` class definition
  - [x] `InputActionPoseSource` serialized field tooltips.
  - [x] `InteractionMode` struct definition and serialized field tooltips and properties
  - [x] `InteractionModeDefinition` serialized field tooltips and properties
  - [x] `InteractionModeManager` public method
  - [x] `InterpolationType` enum values
  - [x] `MRTKHandsAggregatorSubsystem.MRTKAggregator` class definition and public members

## Bugs
- https://github.com/microsoft/MixedRealityToolkit-Unity/issues/11596
- https://github.com/microsoft/MixedRealityToolkit-Unity/issues/11590

### CODE REVIEW

File: com.microsoft.mrtk.input/Interactors/InteractorVisuals/Lines/Definitions/InterpolationType.cs [hunk 1]
```diff
@@ -8,8 +8,14 @@ namespace Microsoft.MixedReality.Toolkit.Input
     /// </summary>
     public enum InterpolationType
     {
+        /// <summary>
+        /// Interpolation using Bézier parametric curve.
+        /// </summary>
         Bezier = 0,
-        CatmullRom,
-        Hermite,
```
Comment (marlenaklein-msft) [inline com.microsoft.mrtk.input/Interactors/InteractorVisuals/Lines/Definitions/InterpolationType.cs:13]: Was Hermite unused?

Comment (marlenaklein-msft) [reply com.microsoft.mrtk.input/Interactors/InteractorVisuals/Lines/Definitions/InterpolationType.cs:13]: I have never heard any of these words before...

**Review by marlenaklein-msft — APPROVED** [review]

File: com.microsoft.mrtk.input/Interactors/InteractorVisuals/Lines/Definitions/InterpolationType.cs (hunk 1 above)
Comment (AMollis) [reply com.microsoft.mrtk.input/Interactors/InteractorVisuals/Lines/Definitions/InterpolationType.cs:13]: Yup :) Unused... see PR description.

Comment (marlenaklein-msft) [reply com.microsoft.mrtk.input/Interactors/InteractorVisuals/Lines/Definitions/InterpolationType.cs:13]: Oop sorry

**Pull request merged**